core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -579,6 +580,35 @@ default CompletableFuture<List<T>> collect(EventExecutor executor, SubscriptionO
         return collector.collect();
     }
 
+    /**
+     * Aggregates the elements published by this {@link StreamMessage} with the specified
+     * {@link AggregationOptions}.
+     *
+     * <p>Example:
+     * <pre>{@code
+     * StreamMessage stream = StreamMessage.of(1, 2, 3, 4);
+     * int sum = stream.aggregate(AggregationOptions.of((nums -> nums.stream().reduce(0, Integer::sum))))
+     *                 .join();
+     * assert sum == 10;
+     * }</pre>
+     */
+    <U> CompletableFuture<U> aggregate(AggregationOptions<T, U> options);
```

Comment:
This looks like a breaking change - I guess we should add a label at least so we don't miss it when writing our release note 😄

Reply:
Add a breaking change label but no one may want to implement `StreamMessage` oneself.